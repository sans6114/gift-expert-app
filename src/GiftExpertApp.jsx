import './App.css';

import { useState } from 'react';

import Heading from '../src/components/Heading';
import { AddCategory } from './components/AddCategory';
import { GridGift } from './components/GridGift';

function GiftExpertApp() {

  const [categories, setCategories] = useState(['valorant'])

  const onAddCategory = (inputValue) => {
    if (categories.includes(inputValue)) return

    setCategories([...categories, inputValue])
  }

  return (
    <main>
      {/* titulo */}
      <Heading title={'Gift Expert App'} subtitle={'An app to improve in REACT library skills'} />
      {/* input */}
      <AddCategory onNewCategory={onAddCategory} />


      {/* listado de gift*/}

      {
        categories.map((categoria) => (
          <GridGift key={categoria} category={categoria} />
        ))
      }
      {/* gift Item*/}




    </main>
  )
}

export default GiftExpertApp
