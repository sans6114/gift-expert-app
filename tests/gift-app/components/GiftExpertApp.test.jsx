import {
  render,
  screen,
} from '@testing-library/react';

import GiftExpertApp from '../../../src/GiftExpertApp';

describe('Test on component <GiftExpertApp />', () => {

const title = 'Gift Expert App'
const subtitle = 'An app to improve in REACT library skills'

  test('Debe de hacer match con el snapshot', () => {
    

    const {container} = render(<GiftExpertApp />)
    expect(container).toMatchSnapshot()
    
  });

  test('Debe de mostrar el componente heading con el titulo, descripcion exacto', () => {
   const {container} = render(<GiftExpertApp/>)

   const heading = screen.getByRole('heading', {level: 1})
   const description = container.querySelector('p');
   expect(heading).toBeTruthy()
   expect(description).toBeTruthy()
   expect(heading.textContent).toContain(title);
   expect(description.textContent).toContain(subtitle);
  });

})