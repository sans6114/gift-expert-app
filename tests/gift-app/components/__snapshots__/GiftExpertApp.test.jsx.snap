// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on component <GiftExpertApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <main>
    <h1
      class="mb-4 text-4xl font-extrabold leading-none tracking-tight text-gray-900 md:text-5xl lg:text-6xl dark:text-white"
    >
      Gift Expert App
    </h1>
    <p
      class="mb-6 text-lg font-normal text-gray-500 lg:text-xl sm:px-16 xl:px-48 dark:text-gray-400"
    >
      An app to improve in REACT library skills
    </p>
    <div
      class="mb-6"
    >
      <form
        aria-label="form"
        class="flex gap-x-2"
      >
        <input
          class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
          id="default-input"
          placeholder="Search category"
          type="text"
          value=""
        />
        <button
          class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
          type="submit"
        >
          <span
            class="flex items-stretch transition-all duration-200 rounded-md px-4 py-2 text-sm"
          >
            Search
          </span>
        </button>
      </form>
    </div>
    <button
      class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
      type="button"
    >
      <span
        class="flex items-stretch transition-all duration-200 rounded-md px-4 py-2 text-sm"
      >
        <span
          aria-label="Spinner button example"
          role="status"
        >
          <svg
            class="inline animate-spin text-gray-200 fill-cyan-600 dark:text-gray-600 h-4 w-4"
            fill="none"
            viewBox="0 0 100 101"
          >
            <path
              d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
              fill="currentColor"
            />
            <path
              d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
              fill="currentFill"
            />
          </svg>
        </span>
        <span
          class="pl-3"
        >
          Loading...
        </span>
      </span>
    </button>
  </main>
</div>
`;
